lua/nvim-autopairs/completion/handlers.lua

```diff
@@ -0,0 +1,78 @@
+local autopairs = require('nvim-autopairs')
+local utils = require('nvim-autopairs.utils')
+
+local M = {}
+
+---@param char string
+---@param item table
+---@param rules table
+---@param bufnr number
+M["*"] = function(char, item, rules, bufnr)
+    local line = utils.text_get_current_line(bufnr)
+    local _, col = utils.get_cursor()
+    local char_before, char_after = utils.text_cusor_line(line, col, 1, 1, false)
+
+    if char == '' or char_before == char or char_after == char
+        or (item.data and item.data.funcParensDisabled)
+        or (item.textEdit and item.textEdit.newText and item.textEdit.newText:match "[%(%[%$]")
+        or (item.insertText and item.insertText:match "[%(%[%$]")
+    then
+        return
+    end
+
+    local new_text = ''
+    local add_char = 1
+
+    for _, rule in pairs(rules) do
```

Comment:
we don't need to check here it still check when you feedkey

Reply:
so I don't need to pass `rules` in handler function?

Reply:
yeah if you don't have any purpose to check insert a matching pair on there. after you call a feedkey it still checks again on mapping.